Integration tests fail due to timeout

### What should be done?

Integration tests sometimes fail with `timeout: context already done: context deadline exceeded`, the failing tests are not always the same.

Failed CI https://github.com/FerretDB/FerretDB/actions/runs/13667937366/job/38212753464?pr=4854
```
2025-03-05T03:38:22.9682792Z WARN       FAIL TestQueryArrayCompatSize/Zero/TestQueryArrayCompatSize_DocumentsStrings (0.12s/10.25s):
2025-03-05T03:38:22.9683326Z WARN       === RUN   TestQueryArrayCompatSize/Zero/TestQueryArrayCompatSize_DocumentsStrings
2025-03-05T03:38:22.9683540Z WARN           query_array_compat_test.go:68: 
2025-03-05T03:38:22.9684079Z WARN                       Error Trace:    /home/runner/work/FerretDB/FerretDB/integration/query_compat_test.go:135
2025-03-05T03:38:22.9684305Z WARN                       Error:          Received unexpected error:
2025-03-05T03:38:22.9686217Z WARN                                       (InternalError) [msg_createindexes.go:67 handler.(*Handler).MsgCreateIndexes] [pool.go:83 documentdb.(*Pool).Acquire] [pool_new.go:143 documentdb.newPgxPoolCheckConn] timeout: context already done: context deadline exceeded
2025-03-05T03:38:22.9686752Z WARN                       Test:           TestQueryArrayCompatSize/Zero/TestQueryArrayCompatSize_DocumentsStrings
2025-03-05T03:38:22.9687321Z WARN       --- FAIL: TestQueryArrayCompatSize/Zero/TestQueryArrayCompatSize_DocumentsStrings (10.25s)
```

### Where?


https://github.com/FerretDB/FerretDB/tree/main/integration

### Definition of Done

- integration/compatibility tests added/updated;
- spot refactorings done;

--- a/internal/documentdb/pool_new.go
+++ b/internal/documentdb/pool_new.go
@@ -55,10 +55,14 @@ func newPgxPool(uri string, l *slog.Logger, sp *state.Provider) (*pgxpool.Pool, 
 	// versions and parameters could change without FerretDB restart
 	config.AfterConnect = func(ctx context.Context, conn *pgx.Conn) error {
 		// see https://github.com/jackc/pgx/issues/1726#issuecomment-1711612138
-		ctx, cancel := context.WithTimeout(ctx, 2*time.Second)
+		ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
 		defer cancel()
 
-		return newPgxPoolCheckConn(ctx, conn, l, sp)
+		if e := newPgxPoolCheckConn(ctx, conn, l, sp); e != nil {
+			return lazyerrors.Error(e)
+		}
+
+		return nil
 	}
 
 	// port tracing, tweak logging

